src/snowflake/snowpark/_internal/ast_utils.py

```diff
@@ -346,84 +370,98 @@ def set_builtin_fn_alias(ast: proto.Expr, alias: str) -> None:
+        # setting src fields for explicit presence of the encapsulating message in the AST.
+        # e.g., Null values have no fields, so the assignment to src fields ensures their presence.
+        if frame is None:
+            src.file = "<unknown>"
+            return expr_ast
+        
+        # NOTE: The inspect module provides many other APIs to get information about the current frame and its callers.
+        # All of these (except for currentframe) unnecessarily incur the overhead of resolving the filename for each
+        # frame and making sure the file exists (even if the context parameter is set to 0 to avoid capturing lineno, 
+        # source line, etc.). Since we should know exactly how many frames to step back, we can and should avoid this 
+        # overhead. Using the inspect.currentframe() (a wrapper around sys._getframe that handles the None case) is the
+        # most efficient way to get the current frame. Stepping back from this frame via frame.f_back is also the most 
+        # efficient method to walk the stack as each frame object contains the minimal amount of needed context.
+        if caller_frame_depth is None:
+            # If the frame is not None, one guarantee we have is that two frames back is the caller's caller, and this
+            # frame contains the code of interest from the user if they are using a simple public API with no further
+            # nesting or indirection. This is the most common case.
+            frame, prev_frame = frame.f_back.f_back, frame.f_back
+            while frame is not None and SNOWPARK_LIB_PATH in Path(frame.f_code.co_filename).parents:
+                frame, prev_frame = frame.f_back, frame
+        else:
+            # If the frame is not None, use the provided stack depth to step back to the relevant frame.
+            # This frame should be the first frame outside of the snowpark package, and contain the code of interest.
+            curr_frame_depth = 0
+            while frame is not None and curr_frame_depth < caller_frame_depth:
+                frame, prev_frame = frame.f_back, frame
+                curr_frame_depth += 1
+
+        if debug:
+            last_snowpark_file = prev_frame.f_code.co_filename
+            assert SNOWPARK_LIB_PATH in Path(last_snowpark_file).parents
+            first_non_snowpark_file = frame.f_code.co_filename
+            assert SNOWPARK_LIB_PATH not in Path(first_non_snowpark_file).parents
+
+        # Once we've stepped out of the snowpark package, we should be in the code of interest.
+        # However, the code of interest may execute in an environment that is not accessible via the filesystem.
+        # e.g. Jupyter notebooks, REPLs, calls to exec, etc.
+        filename = frame.f_code.co_filename if frame is not None else "<unknown>"
+        if frame is None or not Path(filename).is_file():
+            src.file = "<unknown>"
```

Comment:
I'd leave the file as an empty string and let whoever renders the UI decide what to do.